### DISCUSSION TITLE
Confirmation of Retrigger System with use of HID Inputs

### DISCUSSION THREAD

**Discussion by pizman** [MobiFlight/MobiFlight-Connector#557] [Q&A]
[https://www.mobiflight.com/forum/message/24287.html](url)

We need to check if the "Retrigger" Logic also work for Input Configs that use a HID Joystick Button.
Does the logic only work in combination with "Firmware" and Mobiflight Boards..... OR.... Does it also work with the Information of a pressed or released Button  ( e.g. on a Bodnar)

**Comment by ssewell** [comment]
That's an interesting question. Any HID "synchronization" (for lack of a better descriptor) would have to reside in the MF Connector, not the MF Arduino Firmware. I personally haven't reviewed any of the recently added HID features, so I can't really speak to the current implementation.

**Comment by pluckas** [comment]
Retrigger logic works with Bodnar BBI-32 used as an input module in Mobiflight with some limitations - ie, it does not appear to re-trigger 3 way toggle switches configured as 0 - 1 - 2, with 1 connected to ground. 

Interestingly, the physical button mapped to the re-trigger event does not need to be pressed to re-trigger buttons connected to a Bodnar board. They re-trigger automatically when Mobiflight is started with sim running.

**Reply by pizman** [reply]
Additional Request/thought.....

As pluckas request in the Topic....   May we should think about a OPTIONAL   Auto-Retrigger Feature ( Choosable Checkbox in Settings Tab.)
If activated, then Mobiflight automaticly after pressing RUN  start a "Retrigger Request"   same way as we would do it by pressing a assigned Button.

Sounds logical to me.....  A USer who not use Retrigger till now simpla not enable this..... the Retrigger Guys would no longer need this Fake Button ( that would be used pretty sure also on every MF Start by hand) .

A Benefit without Disadvantage !


EDIT:    And by the way.....  Not miss to doublecheck this feature also in next release when we put e.g. Input Shifters or Multiplexer on the screen.    The Retrigger "sould" work for all possible "Button" Devices  ( whatever source) we support !

**Reply by MobiFlight-Admin** [reply]
auto retrigger sounds really like a great idea

**Reply by pluckas** [reply]
No update on the retrigger functionality for some time, so I'm just bumping this up the list again. While auto retrigger is a great idea, the bigger issue remains with the limitation with 3-way toggle switches configured as 0 - 1 - 2, with 1 connected to ground.
